lib/postgrex/extensions/calendar.ex

```diff
@@ -132,25 +134,64 @@ if Code.ensure_loaded?(Calendar) do
 
     # Time
 
-    defp encode_time(_, %Time{microsecond: {microsec, _}} = time) do
+    defp encode_time(type_info, value) do
+      case time_to_microseconds(value) do
+        microsec when is_integer(microsec) ->
+          <<microsec :: int64>>
+        :error ->
+          raise ArgumentError,
+          Postgrex.Utils.encode_msg(type_info, value, Time)
+      end
+    end
+
+    defp time_to_microseconds(%Time{microsecond: {microsec, _}} = time) do
       sec =
         time
         |> Time.to_erl()
         |> :calendar.time_to_seconds()
 
-      <<sec * 1_000_000 + microsec :: int64>>
-    end
-    defp encode_time(type_info, value) do
-      raise ArgumentError,
-        Postgrex.Utils.encode_msg(type_info, value, Time)
+      sec * 1_000_000 + microsec
     end
 
     defp decode_time(<<microsec :: int64>>) do
+      microseconds_to_time(microsec)
+    end
+
+    defp microseconds_to_time(microsec) do
       sec = div(microsec, 1_000_000)
       microsec = rem(microsec, 1_000_000)
       sec
       |> :calendar.seconds_to_time()
       |> Time.from_erl!(microsec)
     end
+
+    # Time with time zone
+
+    @dms_max (:calendar.time_to_seconds({23, 59, 59}) + 1) * 1_000_000
+
+    defp encode_timetz(type_info, value) do
+      case time_to_microseconds(value) do
+        microsec when is_integer(microsec) ->
+          <<microsec :: int64, 0 :: int32>>
+        :error ->
+          raise ArgumentError,
+          Postgrex.Utils.encode_msg(type_info, value, Time)
+      end
+    end
+
+    defp decode_timetz(<<microsec :: int64, tz :: int32>>) do
+      microsec
+      |> timetz_to_microseconds(tz)
+      |> microseconds_to_time()
+    end
+
+    defp timetz_to_microseconds(microsec, tz) do
+      case microsec + tz * 1_000_000 do
+        microsec when microsec < 0 ->
```

Comment:
Oh, microsec is rebound to microsec + tz and tz can make it negative, right? But `microsec` that comes as an argument is never negative. Maybe it would be clearer to use a new name for the result of microsec + tz.